### DISCUSSION TITLE
How to restart kube-proxy?

### DISCUSSION THREAD

**Discussion by J1a-wei** [rancher/rke2#5910] [Q&A] [answered]
Hello RKE team,
We experienced a network outage in our production data center, which lasted for 30 minutes. Although the network is now back to normal, we have noticed that some kube-proxy instances have abnormal conditions, and we are unable to view the logs. 

![image](https://github.com/rancher/rke2/assets/49090950/99ec0819-52d5-414d-a521-73b22a3b90d0)

```
kubectl get po -o yaml kube-proxy-rke-worker-03
Error from server (NotFound): pods "kube-proxy-rke-worker-03" not found
```

I tried to delete it but couldn't, it remains in the Terminating state.
I plan to restart it, but I can't find a way.

Also, for the kube-proxy on worker-03, although it's in an abnormal state, I created an nginx pod and bound it to worker-03. Accessing its service still works.
So kube-proxy may still be working.

We are using the Cilium plugin with RKE2 version 1.26.
How should we proceed?

**Comment by brandond** [comment, marked as answer]
If the node is in an indeterminate state and containers are not manageable, I would probably recommend draining/cordoning it, followed by a reboot of the host. You could try just running the `rke2-killall.sh` script and then restarting rke2, but a reboot may be preferable.

For root cause, you might look at the kubelet and containerd logs around the time where the pod went into the terminating state.

**Reply by fabionitto** [reply]
what if the draining doesn't work? 
Running rk2-killall.sh with bound pvc will remove data?

**Reply by brandond** [reply]
it'll unmount it. it should not delete data from the pvc.
